src/target/riscv/riscv.h

```diff
@@ -129,6 +129,22 @@ struct reg_name_table {
 	char **reg_names;
 };
 
+typedef struct riscv_mem_access_info {
+	target_addr_t address;
+
+	union {
+		uint8_t *read;
+		const uint8_t *write;
+	} buffer;
+
+	uint32_t size;
+	uint32_t increment;
+	uint32_t count;
+
+	bool is_virtual;
```

Comment:
If I read the code correctly, there is no function that would read the  `is_virtual` item.

Also it looks to me the `is_virtual` attribute is not relevant on all hierarchy levels of the memory r/w functions.

For that reason, I'd recommend to remove it for now:

```suggestion
```

If "is_virtual" becomes relevant in a later merge request, then it can be added. But perhaps not as a member of this structure, but more likely as an additional function argument, and only on those functions for which it really matters:

```c
int some_read_write_func(struct target *target, const riscv_mem_access_args_t args, bool is_virtual);
```

Reply:
Addressed